aaelf64-morello/aaelf64-morello.rst

```diff
@@ -759,6 +773,20 @@ Dynamic Morello relocations
 
   ``| 64-bits offset | 64-bits size |``
 
+  ``R_MORELLO_CODE_CAPINIT`` is similar to ``R_MORELLO_CAPINIT`` but signifies
+  that the capability being relocated is a code pointer as opposed to a data
+  pointer or function pointer. This distinction is needed for library-based
+  compartmentalization (c18n). In practice, this relocation type should only be
+  seen in intermediate relocatable objects and not in executables or shared
+  objects because the toolchain should only allow code pointers to be taken for
+  non-preemptible symbols. These relocations would subsequently be converted to
+  relative relocations at static-link time.
```

Comment:
Do we want to explicitly state that this restriction may be enforced? (Or make it a static relocation, but that would mean we could never backpedal on this if it turns out we need it for preemptible symbols...)

Reply:
I put this explanation here mainly so that nobody can get confused over why RTLD doesn't actually handle this relocation type.

How about I change the language a bit to say that the current behaviour of the toolchain is to not allow preemptible symbols, but this may change in the future.

Reply:
I've reworked the language.